### DISCUSSION TITLE
shadow problem

### DISCUSSION THREAD

**Discussion by tjzzhang** [bnpr/Malt#476] [Q&A]
![problem](https://user-images.githubusercontent.com/9721816/214791372-e63d093f-21af-48da-b1c0-1fdc0ca3a9ba.png)
The partially transparent object's shadow is not correct, how to fix the problem? thanks

**Comment by eballeste** [comment]
has been asked before

https://github.com/bnpr/Malt/discussions/337

**Reply by tjzzhang** [reply]
It helps, Thank you!

**Reply by tjzzhang** [reply]
![无标题](https://user-images.githubusercontent.com/9721816/214852085-6364f9d2-c943-474f-a894-d5d09d9ea792.png)
The shadow is ok, but when I press F12, it can't render the picture. 
The log says:

Malt > Texture Uniform SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_POINT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_POINT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_POINT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_POINT not found
Malt > Texture Uniform IN_LIGHT_CUSTOM_SHADING not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_POINT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_POINT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_POINT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_DEPTH_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_ID_POINT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SPOT not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_SUN not found
Malt > Texture Uniform TRANSPARENT_SHADOWMAPS_COLOR_POINT not found
Malt > Texture Uniform IN_LIGHT_CUSTOM_SHADING not found
Malt > STATS: Viewport (1):
Resolution : (1304, 677)
Sample : 9 / 9
Sample Time : 73.775 ms
Total Time : 0.664 s
Latency : 3 frames
Max Latency : 3 frames

**Reply by eballeste** [reply]
gotcha, you should [report this as a possible bug](https://github.com/bnpr/Malt/issues/new?assignees=&labels=&template=bug_report.yml).

**Reply by tjzzhang** [reply]
I have reported, but there is no response at present. 
The problem is very strange. It can be rendered by changing to bmp format, but failed again when changing the camera 's angle.
